lab1.js

```diff
@@ -106,19 +124,27 @@ assert(nextAnimal, 'assign something to nextAnimal');
 */
 
 // number of times the new caretaker fed the lion. one array entry per day
-var mealsPerDay = [5, 4, 3, 6, 2, 4, 3, 4, 5, 1];
+var mealsPerDay = [ 5, 4, 3, 6, 2, 4, 3, 4, 5, 1 ];
 var tooHungryDay;
 
-/*
- TODO: 20 points
- Cycle through the days in mealsPerDay. At each day, print out the average
- number of meals/day the lion got since the new caretaker started.
- tooHungryDay should receive the number of days before the lion started
- pondering protein supplements (the first day the average dips below 4
- meals)
-*/
+var mealsTotal = 0;
+var avgMealsDay = Math.round(mealsTotal / (i + 1) * 100) / 100;
+var avgMealsDayArr = [];
 
-// your code goes here
+for (var i = 0; i < mealsPerDay.length; i++) {
+  mealsTotal += mealsPerDay[i];
+  avgMealsDay = Math.round(mealsTotal / (i + 1) * 100) / 100;
+  avgMealsDayArr.push(avgMealsDay);
+
+  console.log('The lion has eaten ' + avgMealsDay + ' meals on average.');
+}
+
+for (var i = 0; i < avgMealsDayArr.length; i++) {
+  if (avgMealsDayArr[i] < 4) {
+    tooHungryDay = i;
+    break;
```

Comment:
good use of break here. No need to run the rest of the loop if you don't need it.